### PR TITLE
tests(parameters): integration tests for `SecretsProvider`

### PR DESCRIPTION
**Pull request by am29d**
## Description of your changes

As part of #1039 , we need to implement integration tests for the SecretsProvider which is part of the upcoming Parameters utility. 

The test cases cover string, json, binary and auto transformer. We don't cover the cases for sdk client options  and I will add them after #1222 is resolved.

### Related issues, RFCs

**Issue number:** #1241 

### PR status

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by am29d** [comment]
Should I keep it as a draft util we merge #1260 for all types?

File: packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts
Comment (dreamorosi) [inline packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts:4]: Why do we import this and use it in the conditional? Not challenging, just asking because I'm not clear on it

File: packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts
Comment (dreamorosi) [inline packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts:16]: How will this work once we introduce multiple providers (i.e. once #1222 is merged and we need multiple SecretsProviders)? Should we add a new parameter?

**Review by dreamorosi — COMMENTED** [review]
Thank you for the PR, appreciate it!

Left a couple of minor comments to understand the function code.

Happy to merge this once review is ready, no need to wait for other PRs.

Whenever you have time, could you please leave a comment under the linked issue so that I can assign it to you?

File: packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts
Comment (am29d) [reply packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts:4]: This does not make any sense 🤔 , it should be `options` in the conditional. Will change that.

File: packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts
Comment (am29d) [reply packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts:16]: Right, lets pass the provider in the signature. Should we wait until #1222 is merged, or can I change it now?

Comment (dreamorosi) [reply packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts]: Let's do it now, so later we can just add a new block at the end of the main handler

Comment (shdq) [reply packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts]: Hey guys! #1222 is ready for review.

**Comment by dreamorosi** [comment]
Hi @am29d I have merged #1222 earlier today, so the tests that involve caching and/or passing a custom AWS SDK client can now be worked on.

Do you prefer adding them to this PR, or should we open a separate issue to address them in a future PR?

I'm okay either way.

**Comment by am29d** [comment]
I added two test cases, one for caching and other for force fetch. I am not sure what other cases with custom SDK client we should cover with e2e tests. The major benefit comes from injecting clients for testing, which is not something we should asses with e2e. 

What do you think?

**Comment by dreamorosi** [comment]
> What do you think?

Completely agree, the fact that we can check the caching already proves the injection works correctly. I'm okay with the current test surface for now.

**Comment by dreamorosi** [comment]
I have ran the integration tests locally after checking out your branch and they are passing:

Screenshot 2023-02-07 at 17 06 43

**Review by dreamorosi — APPROVED** [review]

**Comment by dreamorosi** [comment]
Thank you for the work on this Alex, really appreciate it!

**Pull request merged**
